src/queue.cpp

```diff
@@ -297,21 +299,36 @@ cl_int cvk_command_group::execute_cmds() {
                      cl_command_type_to_string(cmd->type()), cmd->event());
 
         cl_int status = cmd->execute();
-        if (status != CL_COMPLETE && global_status == CL_SUCCESS)
+        if (status != CL_COMPLETE && global_status == CL_SUCCESS) {
             global_status = status;
+        }
         cvk_debug_fn("command returned %d", status);
 
         commands.pop_front();
 
-        // Deleting batch with many commands can take a while. Trace it to be
-        // able to understand it easily.
+        // Deleting batch with many commands can take a while. Trace it
+        // to be able to understand it easily.
         TRACE_BEGIN("delete_cmd");
         delete cmd;
         TRACE_END();
     }
     return global_status;
 }
 
+bool cvk_command_queue::get_printf_callback(printf_callback_func& callback) {
+    auto& properties = m_context->properties();
+    for (unsigned i = 0; i < properties.size(); i += 2) {
+        if (properties[i] == CL_PRINTF_CALLBACK_ARM) {
+            void (*func_ptr)(const char*, long unsigned int) =
```

Comment:
I think it should be possible to use the `printf_callback_func` type here somehow.